packages/react-aria-components/docs/DateRangePicker.mdx

```diff
@@ -0,0 +1,1120 @@
+
+```tsx example
+function Example() {
+  let [date, setDate] = React.useState({
+    start: parseAbsoluteToLocal('2021-04-07T18:45:22Z'),
+    end: parseAbsoluteToLocal('2021-04-08T20:00:00Z')
+  });
+
+  return (
+    <>
+      <MyDateRangePicker
+        label="Date and time range"
+        granularity="second"
+        value={date}
+        onChange={setDate} />
+      <MyDateRangePicker
+        label="Date range"
+        granularity="day"
+        value={date}
+        onChange={setDate} />
+    </>
+  );
+}
+```
+
+
+If no `value` or `defaultValue` prop is passed, then the `granularity` prop also affects which type of value is emitted from the `onChange` event. Note that by default, time values will not have a time zone because none was supplied. You can override this by setting the `placeholderValue` prop explicitly. Values emitted from `onChange` will use the time zone of the placeholder value.
+
+```tsx example
+import {now} from '@internationalized/date';
+
+<MyDateRangePicker
+  label="Date range"
+  granularity="second" />
+<MyDateRangePicker
+  label="Date range"
+  placeholderValue={now('America/New_York')}
+  granularity="second" />
+```
+
+### International calendars
+
+`DateRangePicker` supports selecting dates in many calendar systems used around the world, including Gregorian, Hebrew, Indian, Islamic, Buddhist, and more. Dates are automatically displayed in the appropriate calendar system for the user's locale. The calendar system can be overridden using the [Unicode calendar locale extension](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/calendar#adding_a_calendar_in_the_locale_string), passed to the [I18nProvider](I18nProvider.html) component.
+
+Selected dates passed to `onChange` always use the same calendar system as the `value` or `defaultValue` prop. If no `value` or `defaultValue` is provided, then dates passed to `onChange` are always in the Gregorian calendar since this is the most commonly used. This means that even though the user selects dates in their local calendar system, applications are able to deal with dates from all users consistently.
+
+The below example displays a `DateRangePicker` in the Hindi language, using the Indian calendar. Dates emitted from `onChange` are in the Gregorian calendar.
+
+```tsx example
+import {I18nProvider} from '@react-aria/i18n';
+
+function Example() {
+  let [range, setRange] = React.useState(null);
+  return (
+    <I18nProvider locale="hi-IN-u-ca-indian">
+      <MyDateRangePicker label="Date range" value={range} onChange={setRange} />
+      <p>Start date: {range?.start.toString()}</p>
+      <p>End date: {range?.end.toString()}</p>
+    </I18nProvider>
+  );
+}
+```
+
+### Minimum and maximum values
```

Comment:
we need more than color to indicate that the range is invalid, add an icon? or a helptext?

Reply:
I marked this down for followup. It's not so straightforward to make the extra icon appear/disappear based on state.  We could wrap the whole DateRangePicker children in a function so we can access the validation state but it does increase the complexity of the example a lot. We could expose a new data attribute on the DateRangePicker and hide/show via CSS also but not sure this is a good idea...